### DISCUSSION TITLE
Streams and Custom Statements

### DISCUSSION THREAD

**Discussion by mik3lov3st3ch** [simolus3/drift#2867] [Q&A]
I use .watch() on select statements to receive updates as a stream. That works fine as long as the Drift API for mutation ist used. As soon as I use custom statements, the stream doesn´t emit a new value.

As I have to use custom statements in some scenarios (e.g. incrementing a column value) I´m looking for a way to maually emit new values to the stream listeners after I´ve executed a custom statement.

Is there a way to handle this?

**Comment by simolus3** [comment]
When using custom select statements, you need to specify the tables you're selecting from manually to get updates. You can do that with the `readsFrom` parameter in `customSelect`. Similarly, `customUpdate` has an `updates` parameter to signal which tables are being updated.

An alternative is to use [drift files](https://drift.simonbinder.eu/docs/sql-api/drift_files/), which lets you get your custom SQL checked at build time. It also enables drift to figure out which tables are being watched by the query.

> As I have to use custom statements in some scenarios (e.g. incrementing a column value)

This can actually be done in drift with the `.custom` "constructor" on companions, e.g. `update(table).write(TableCompanion.custom(counter: table.counter + Variable(1)))`.
